src/components/gallery/GalleryPage.vue

```diff
@@ -0,0 +1,62 @@
+<script setup lang="ts">
+import { backendManager } from '@/Backend';
+import { useGenesisStore } from '@/builder/GenesisStore';
+import { getCurrentNetwork } from '@/chain/Network';
+import Footer from '@/components/landing_page/Footer.vue';
+import Header from '@/components/landing_page/Header.vue';
+import { ref, watch } from 'vue';
+
+const genesisStore = useGenesisStore();
+
+const sort = ref('latest')
+const galleryData = ref<Object[]>([])
+const loading = ref(true)
+
+watch([sort], () => {
+  fetchGalleryData()
+})
+
```

Comment:
I tended to use `watchEffect` unless explicitly specifying was necessary. This would also cover e.g. a network change.